### PR TITLE
[Model] Add jobs to github action model.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Few things need documentation. I need to see how they'll work. Specially with the If condition.

### CODE REVIEW

File: src/Sharpliner/GitHubActions/Environment.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    /// <summary>
+    ///
+    /// </summary>
+    public record Environment
+    {
+        public Environment(string name, string? url = null)
+        {
+            if (string.IsNullOrEmpty(name))
+                throw new ArgumentNullException(nameof (name));
+            Name = name;
```
Comment (premun) [inline src/Sharpliner/GitHubActions/Environment.cs:14]:
```suggestion
            Name = name ?? throw new ArgumentNullException(nameof(name));
```

File: src/Sharpliner/GitHubActions/Job.cs [hunk 2]
```diff
@@ -1,7 +1,65 @@
...
+
 namespace Sharpliner.GitHubActions
 {
     public record Job
     {
+        public Job(string id)
+        {
+            if (string.IsNullOrEmpty(id))
+                throw new ArgumentNullException((nameof (id)));
+            Id = id;
```
Comment (premun) [inline src/Sharpliner/GitHubActions/Job.cs:12]:
```suggestion
            Id = id ?? throw new ArgumentNullException((nameof(id)));
```

**Review by premun — APPROVED** [review]

File: src/Sharpliner/GitHubActions/Environment.cs (hunk 1 above)
Comment (mandel-macaque) [reply src/Sharpliner/GitHubActions/Environment.cs:14]: But like that it can be empty :)

File: src/Sharpliner/GitHubActions/Job.cs (hunk 2 above)
Comment (mandel-macaque) [reply src/Sharpliner/GitHubActions/Job.cs:12]: Same, it can be empty

Comment (premun) [reply src/Sharpliner/GitHubActions/Job.cs:12]: Sorry, gotcha!

Comment (premun) [reply src/Sharpliner/GitHubActions/Job.cs:12]: Wait

Comment (premun) [reply src/Sharpliner/GitHubActions/Job.cs:12]: What is the usecase?

Comment (premun) [reply src/Sharpliner/GitHubActions/Job.cs:12]: Just more strict validation?

**Pull request merged**